l2geth/miner/worker.go

```diff
@@ -1172,13 +1173,16 @@ func (w *worker) commitNewTx(tx *types.Transaction, txSetProof *types.BatchTxSet
 		meta.Index = &index
 		tx.SetTransactionMeta(meta)
 	}
+	latestRollbackState := w.eth.BlockChain().LatestRollbackStates()
 	header := &types.Header{
-		ParentHash: parent.Hash(),
-		Number:     new(big.Int).Add(num, common.Big1),
-		GasLimit:   w.config.GasFloor,
-		Extra:      w.extra,
-		Time:       tx.L1Timestamp(),
-		Coinbase:   txSetProof.Sequencer,
+		ParentHash:     parent.Hash(),
+		Number:         new(big.Int).Add(num, common.Big1),
+		GasLimit:       w.config.GasFloor,
+		Extra:          w.extra,
+		Time:           tx.L1Timestamp(),
+		Coinbase:       txSetProof.Sequencer,
+		RollbackIndex:  latestRollbackState.Index,
```

Comment:
Is there a better way to achieve the goal? Do not change this structure, because most tools of eth will have a method to obtain the header, and this change may let it incompatibility.